webdriver-spec.html

```diff
@@ -4581,14 +4583,35 @@
  the following steps need to be completed:
 
 <ol>
- <li><p>Let <var>result</var> be Return the result of
+ <li><p>Let <var>evaluateResult</var> be the result of
   calling <a><code>evaluate</code></a>, with
-  arguments <var>selector</var>, <a>context object</a> equal to
-  the <var>start node</var>.
+  arguments <var>selector</var>, <var>start node</var>,
+  <a><code>null</code></a>, <a>ORDERED_NODE_SNAPSHOT_TYPE</a>, and
+  <a><code>null</code></a>.
+
+  <p class=note>A snapshot is used to promote operation atomicity.
+
+ <li><p>Let <var>index</var> be 0.
+
+ <li><p>Let <var>length</var> be the result of <a>getting the property</a>
+  "<code>snapshotLength</code>" from <var>evaluateResult</var>.
+
+ <li><p>Let <var>result</var> be an empty <a>NodeList</a>.
 
- <li><p>If any item in <var>result</var> is not an <a>element</a>
-  return an <a>error</a> with <a>error code</a> <a>invalid
-  selector</a>.
+ <li><p>Repeat, while <var>index</var> is less than <var>length</var>:
+
+  <ol>
+   <li>Let <var>node</var> be the result of calling <a>snapshotItem</a> with
+    argument <var>index</var>, with the <a>context object</a> equal to
+    <var>evaluateResult</var>.
+
+   <li><p>If <var>node</var> is not an <a>element</a> return an <a>error</a>
+    with <a>error code</a> <a>invalid selector</a>.
```

Comment:
The error handling in [the call site](https://w3c.github.io/webdriver/webdriver-spec.html#dfn-find) is a little ambiguous:

> 6. If a DOMException, SyntaxError, XPathException, or other error occurs
>    during the execution of the element location strategy, return error
>    invalid selector. 

The intention is clear, but to the pedantically-minded (like me), it's not quite accurate to say that an error "occurs" in this case. I think it would be better if we were to explicitly "throw" in this patch. There is no convention for throwing within the WebDriver specification, so I could use an editor's input on that.

Alternatively, we could update "Find" with something like, "if an error is thrown or returned by the previous step."